d3fc/d3fc
# Repository Caretaking

# Monorepo Cleanup PR

This PR attempts to clean up some of the miscellaneous leftovers from when the repository was combined into a monorepo and to normalize the procedures for building, bundling and testing the packages within.

_Unfortunately this has become quite a large PR so I will provide a summary of changes below:_

1. __Remove `d3fc-scripts`.__
    This package was unusual in that it was a package of script files which generated the build and perform tests. Given that we are now in a monorepo, there was no particular need for this. It has been removed.
      * The `bundle.js` script has been replaced with using the `rollup` cli.
      * The `required` folder has not been replaced, I would suggest any file requirements be enforced by travis.
      * The `lint.js` script has been replaced with using the `eslint` and `markdownlint` cli
      * The `screenshot.js` script has been moved to a `util` folder.

2. __Add `script` folder__
   A folder has been added to the project root (as opposed to the packages directory) which contains key scripts and utilities used by all projects. This includes 
      * `jasmine.json` test config file 
      * `rollup.config.js` bundle config file 
      * The aforementioend `screenshot.js` screenshotting tool

3. __Remove duplicate files__
   The files `.npmignore`, `.gitignore`, `.eslintrc.json`, and `LICENSE` were duplicated with the same content in each package. A copy of each of these files has been moved to the project root, and their duplicates removed. 

4. __Enable Hot Module Replacement and Realtime Dev Mode__
   This PR also enables a better developer environment. After running `npm i` from the project root, you can call `npm start` which will bundle the packages and launch a developer sandbox in your default web browser which will automatically refresh whenever you save a change to any of the packages. See below:
![demo](https://user-images.githubusercontent.com/19157374/62783935-3f93ab80-bab5-11e9-8bf4-1c5037e16d4b.gif)
